src/conversation_engine/engine.py

```diff
@@ -162,17 +164,19 @@ def __init__(self, robot_name, grammar, command_target):
         self._grammar = grammar
         self._command_target = command_target
 
-        self._user_to_robot_sub = rospy.Subscriber("user_to_robot", String, self._handle_user_to_robot)
-        self._robot_to_user_pub = rospy.Publisher("robot_to_user", String, queue_size=10)
+        self.give_examples = give_examples
```

Comment:
Why is this not private?

Reply:
Because it's fine if a user of this class modifies it. ConversationEngine does not set it anywhere, just reads it when needed.